### ISSUE TITLE
Can a model both use CashierConnectBillable trait and ConnectCustomer?

### ISSUE THREAD

**Issue by Liinkiing** [l4nos/laravel-cashier-stripe-connect#36]
Hello!

First of all, thanks for the package, it looks really promising! I wanted to test it against the app I'm currently developping. It is basically an app where coaches can register coaching packages and invite clients for their training. Then coaches have online calendars in which clients can book a training. 

Technically, I have the same `User` model for the coach and the client. I have a `type` column that can be either `COACH | CLIENT`.

I wanted to use Stripe Connect so coaches are basically Stripe Connected Account, and the clients of the coach will be Connected Customers. But given that I use the same class for both, I can't use both traits in the same class. 

In the beginning of the project, I've thought about trying to have distinct models (even they share a lot), but I've read that it is much harder to work with Laravel auth system when having different models, because Coaches and Clients both are logged user. 

Maybe I'm having the wrong approach, but I don't know how doable would it be to use your package when the Connected Account and the Connected Customer is the same model (`User`)?

Again, thank you for the package and for your response! 🙏

**Comment by l4nos** [comment]
The current model structure doesn't support a user being both a merchant and a customer simultaneously across multiple relationships. Here's why:

Customer scope: A customer is tied to a single connected account, not a general entity across multiple merchants.
Stripe's structure: In Stripe's direct charges model, a customer exists separately within each merchant's connected account.

Proposed solution:

- Create a central User model representing the logged-in user.

- Develop separate Customer and Coach models, each linked to their respective Stripe entities.

- Connect these models back to the User model with a user_id field.

This approach allows a user to exist as both a coach and a customer of multiple other coaches, maintaining the necessary separation in Stripe while preserving the relationships in your application.

Example structure:

User (central entity)

Coach (linked to Stripe Connected Account)
Customer (multiple instances, each linked to a different coach's Stripe Customer)

This structure provides the flexibility needed for users to have multiple roles across different relationships in your application.

**Comment by Liinkiing** [comment]
Yes I've thought of something like this, I think this should get the works done! Thank you for your message! 🙏
